secp256k1jni/src/main/java/org/bitcoin/NativeSecp256k1.java

```diff
@@ -568,6 +568,45 @@ public static boolean isValidPubKey(byte[] pubkey) {
         return retVal == 0 ? new byte[0] : sigArr;
     }
 
+    /**
+     * libsecp256k1 Computes a Schnorr public key given a pre-committed R value.
+     *
+     * @param data Message hash, 32 bytes
+     * @param rKey Pre-committed R value, 33 bytes
+     * @param pubKey Signer's public key, 33 bytes
+     * @return sigPubKey byte array with public key associated with signature
+     */
+    public static byte[] computeSchnorrPubKey(byte[] data, byte[] rKey, byte[] pubKey) throws AssertFailException {
+        checkArgument(data.length == 32 && rKey.length == 33 && pubKey.length == 33);
+
+        ByteBuffer byteBuff = nativeECDSABuffer.get();
+        if (byteBuff == null || byteBuff.capacity() < data.length + rKey.length + pubKey.length) {
+            byteBuff = ByteBuffer.allocateDirect(32 + 33 + 33);
+
+            nativeECDSABuffer.set(byteBuff);
+        }
+        byteBuff.rewind();
+        byteBuff.put(rKey);
+        byteBuff.put(data);
+        byteBuff.put(pubKey);
+
+        byte[][] retByteArray;
+
+        r.lock();
+        try {
+            retByteArray = secp256k1_schnorrsig_compute_pubkey(byteBuff, Secp256k1Context.getContext());
+        } finally {
+            r.unlock();
+        }
+
+        byte[] sigPubKey = retByteArray[0];
+        int retVal = new BigInteger(new byte[] { retByteArray[1][0] }).intValue();
```

Comment:
Could we do a comment on the intention here? Seems very cryptic

Reply:
I would if I understood it :P

Reply:
`secp256k1_schnorrsig_compute_pubkey()` returns a two element array of byte arrays. The first element is a byte array containing the public key, the second one contains just a single byte that represents the error code (`0` means error). Using `BigInteger` here is overkill.
`byte retVal = retByteArray[1][0];` would be more than enough.

Reply:
@nkohen thoughts on @rorp s alternative?

Reply:
@nkohen why no thoughts on this in this debate?